packages/x-components/src/x-modules/search-box/components/search-input.vue

```diff
@@ -231,6 +232,19 @@
     protected emitUserAcceptedAQuery(query: string): void {
       this.$x.emit('UserAcceptedAQuery', query, this.createEventMetadata());
     }
+
+    /**
+     * Prevents the user from typing special characters in the input field.
+     * Checks the input query value and replaces special characters with whitespace.
+     *
+     * @internal
+     */
+    protected preventSpecialKey(): void {
```

Comment:
I understand, so instead of removing the special characters after receiving the input, it would more ideal to prevent if before input I check whether the special characters are typed or not and prevent the input? Also, if I were to change to a prevent method using beforeinput, then should I continue using the @keydown? Thanks!

Reply:
You can remove the `keydown` event and rely on the `beforeinput` only. The `beforeinput` event happens before the value of the `input` is modified, and by calling `event.preventDefault()` in the event handler you can indeed prevent the input from being modified. The `event.data` property contains the new typed or pasted characters, so you can test if this string contains the invalid characters, and then prevent them from being "confirmed".